### PR TITLE
#162948845 Allow users to like or dislike articles

### PR DESCRIPTION
**Pull request by Georgeygigz**
**What does this PR do?**
Implement functionality that allows users to like or dislike an article

**Description of Task to be completed?**
- Add functionality for liking an article
- Add functionality for disliking an article
- Write tests for the implementation

**How should this be manually tested?**

- 'git clone https://github.com/andela/ah-django-unchained'

- `cd ah-django-unchained`

- git checkout ft-like-dislike-article-162948845

- create a virtual env python3 -m venv virtual_env_name

- pip install -r requirements.txt

**Start in postman using the following url**
 - like [put]
```
http://127.0.0.1:8000/api/<slug>/like/
```
- dislike[put]
```
http://127.0.0.1:8000/api/<slug>/dislike/
```

**Pivotal tracker stories**
- [story](https://www.pivotaltracker.com/story/show/162948845)

**Like an article**
image

**Dislike an article**
image

### CODE REVIEW

File: authors/apps/articles/serializers.py [hunk 1]
```diff
@@ -10,6 +10,10 @@ class ArticleSerializer(serializers.ModelSerializer):
     author = serializers.SerializerMethodField()
     # Uploads an image to the Cloudinary servers
     images = serializers.ImageField(default=None)
+    likes = serializers.PrimaryKeyRelatedField(many=True, read_only=True)
```
Comment (d-kahara) [inline authors/apps/articles/serializers.py]: Consider renaming this to something related to what it does perhaps `reactors_id` or `user_id`

Comment (Georgeygigz) [reply authors/apps/articles/serializers.py]: okay noted

File: authors/apps/articles/serializers.py [hunk 2]
```diff
@@ -10,6 +10,10 @@ class ArticleSerializer(serializers.ModelSerializer):
...
+    dislikes = serializers.PrimaryKeyRelatedField(many=True, read_only=True)
```
Comment (d-kahara) [inline authors/apps/articles/serializers.py]: ditto

Comment (Georgeygigz) [reply authors/apps/articles/serializers.py]: okay

**Comment by andrewhingah** [comment]
your base test file is named as `base_tests.py` yet it's for likes and dislikes only. There are other tests related to the article, so consider renaming the file e.g. `likes_base_test.py`

**Comment by andrewhingah** [comment]
> your base test file is named as `base_tests.py` yet it's for likes and dislikes only. There are other tests related to the article, so consider renaming the file e.g. `likes_base_test.py`

Or we can agree to have a base test file for all tests related to articles

**Review by d-kahara — APPROVED** [review]

**Review by andrewhingah — APPROVED** [review]

**Review by KMaina — APPROVED** [review]

File: authors/apps/articles/models.py [hunk 3]
```diff
@@ -18,3 +18,7 @@ class Article(models.Model):
     is_deleted = models.BooleanField(default=False)
     is_published = models.BooleanField(default=False)
     author = models.ForeignKey(User, on_delete=models.CASCADE)
+    user_id_likes = models.ManyToManyField(
+        User, related_name='likes', blank=True)
+    user_id_dislikes = models.ManyToManyField(
+        User, related_name='dislike', blank=True)
```
Comment (ignatiusukwuoma) [inline authors/apps/articles/models.py]: Would you like to use `dislikes` since `likes` is plural?

Comment (Georgeygigz) [reply authors/apps/articles/models.py]: Yeah let me change that

File: authors/apps/articles/urls.py [hunk 4]
```diff
@@ -11,4 +11,6 @@
     path('articles/delete/<slug:slug>/',
          DeleteArticle.as_view(),
          name="articles-delete"),
+    path('<slug>/like/', LikeArticleApiView.as_view(), name='likes'),
+    path('<slug>/dislike/', DislikeArticleApiView.as_view(), name='dislikes')
```
Comment (ignatiusukwuoma) [inline authors/apps/articles/urls.py]: shouldn't we have `articles/` in the endpoint?

File: authors/apps/articles/tests/like_dislike_base_test.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        token = response.data['token']
+        return token
+        
+    def signup_user_two(self):
+        """Register new user who will like or dislike an article."""
+        response = self.client.post(self.signup_url,
+                                    self.user_signup_data2,
+                                    format='json')
+        token = response.data['token']
+        return token
```
Comment (ignatiusukwuoma) [inline authors/apps/articles/tests/like_dislike_base_test.py:38]: Let one function register any user passed in as argument

**Comment by ignatiusukwuoma** [comment]
Rebase your branch

**Pull request merged**